two_factor_authentication/business_logic.py

```diff
@@ -1,166 +1,36 @@
-    cache_delay = 300
-
-    @classmethod
-    def cache_set(cls, user):
-        code_2fa = ''.join(random.choices('123456789', k=6))
-        cache.set(user.id, {'code': code_2fa}, cls.cache_delay)
-        return code_2fa
-
-
-class Auth2FAEmail(Cache2FA):
-    """Class which uses Email for checking or performing Two-factor verification"""
-    auth2fa_type = 'Email'
-    cache_delay = CACHE_TIMEOUT_2FA.get(auth2fa_type)
-
-    @classmethod
-    def _perform_2fa(cls, user):
-        """Method that:
-        1. Checks if 2fa code already exists.
-        2. If the 2fa code doesn't exist, this method will create and send it to email.
-        returns a Response depending on the state of execution."""
-        if cache.get(user.id, None):
-            return cls.already_exists
-
-        code_2fa = cls.cache_set(user)
-        context = {
-            'code_2fa': f'{code_2fa}',
-        }
-        confirmation_email = {
-            'subject': 'Personal code for Two-Factor Authentication',
-            'message': 'Personal code for Two-Factor Authentication',
-            'from_email': os.environ.get('auth_user'),
-            'recipient_list': [user.email],
-            'fail_silently': False,
-            'auth_user': os.environ.get('auth_user'),
-            'auth_password': os.environ.get('email_token'),
-            'html_message': render_to_string('2fa_mail.html', context=context),
-        }
-        send_mail(**confirmation_email)
-
-        return cls._approve_answer()
-
-    @classmethod
-    def _check_2fa(cls, user, received_code):
-        """Method that:
-        1. Checks if system 2fa code exists.
-        2. Checks if system 2fa code identical to the users 2fa code.
-        returns a Response depending on the state of execution or delete
-        system 2fa code."""
-        cache_code = cache.get(user.id, {}).get('code', None)
-        if not cache_code:
-            return cls.code_expired
-        if cache_code != received_code:
-            return cls.not_valid
-
-        cache.delete(user.id)   # 2fa already passed so we deleted code from cache
-
-
-class Auth2FAGAUTH(Base2FA):
-    """Class which uses Google Authentication for checking or performing
-    Two-factor verification"""
-    pass
-
-
-def perform_2fa_request(request):
-    """ An Abstraction that defines the type of Two-factor
-    verification. Performs primary routing depending on the
-    presence of request.header. """
-
-    user = request.user
-    # if user.type_2fa == 'email'
-    Auth2FAClass = Auth2FAEmail
-
-    if not (received_code := request.META.get('HTTP_2FACODE', None)):
-        return Auth2FAClass.perform_2fa(user)
-
-    return Auth2FAClass.check_2fa(user, received_code)
+import pyotp
+
+from two_factor_authentication.models import GoogleAuth
+from two_factor_authentication.serializers import GoogleAuthSerializer
+from Web_Menu_DA import settings
+from Web_Menu_DA.constants import Types2FA
+
+MSG_TPL = 'Two-factor verification {}'
```

Comment:
into constants

Reply:
moved to two_factor_authentication.constants.py